test/standalone/Vegetation/test_pmodel.jl

```diff
@@ -216,3 +223,251 @@ end
         end
     end
 end
+
+@testset "Test P-model callback initialization" begin
+    FT = Float32
+    
+    lat = FT(38.7441) 
+    long = FT(-92.2000) 
+    start_date = DateTime(2025) 
+    dt = 600.0 # 10 minutes
+
+    canopy_domain = Point(; z_sfc = FT(0.0), longlat = (long, lat))
+
+    # Autotrophic respiration 
+    ar_params = AutotrophicRespirationParameters(FT)
+    ar_model = AutotrophicRespirationModel{FT}(ar_params)
+
+    # Radiative transfer 
+    rt_params = BeerLambertParameters(FT)
+    rt_model = BeerLambertModel{FT}(rt_params)
+
+    # Photosynthesis 
+    is_c3 = FT(1) 
+    photosynthesis_params = FarquharParameters(FT, is_c3)
+    photosynthesis_model = FarquharModel{FT}(photosynthesis_params)
```

Comment:
It probably makes more sense to use the PModel for this test :) but it will test what you want either way